File: java/dsframework/src/main/java/dsf/pool/SpecifiedServerClientSource.java
package dsf.pool;

import dsf.register.MsgServiceDefine;
import dsf.register.MsgSvcAddr;
import dsf.register.ServiceInstance;
import java.util.HashMap;
import java.util.LinkedList;
import java.util.List;
import java.util.Map;

/**
 *
 * @author Administrator
 * @param <I>
 */
public class SpecifiedServerClientSource<I> extends ClientSource<I> {

    private final ServiceInstance instance;

    public SpecifiedServerClientSource(String regname, String host, int port, int timeout) {
        super(regname, timeout);
        instance = new ServiceInstance(new MsgSvcAddr(host, port));
        final MsgServiceDefine def = new MsgServiceDefine(regname, regname, "SpecifiedServerClientSource", "", new HashMap<String, String> ());
        instance.init(def);
    }

    @Override
    public RawClientInfo getRowClient() throws Exception {
        RawClientInfo raw = new RawClientInfo();
        raw.instance = this.instance;
        raw.client = creator.createClient(instance, timeout);
        return raw;
    }

    @Override
    public List<RawClientInfo> getRowClients() throws Exception {
        List<RawClientInfo> infos = new LinkedList<RawClientInfo>();
        RawClientInfo raw = new RawClientInfo();
        raw.instance = this.instance;
        raw.client = creator.createClient(raw.instance, timeout);
        infos.add(raw);
        return infos;
    }
}

File: csharp/dsf/dsf/src/pool/SpecifiedServerClientSource.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace com.baidu.dsf.pool
{
    using com.baidu.dsf.adaptor;
    using com.baidu.dsf.register;
    class SpecifiedServerClientSource<I> : ClientSource<I>
    {
        private ServiceInstance instance;


        public SpecifiedServerClientSource(string regname, String host, int port, int timeout) : base(regname, timeout)
        {
            instance = new ServiceInstance(new MsgSvcAddr(host, port));
        }

        public override RawClientInfo getRowClient()
        {
            RawClientInfo raw = new RawClientInfo();
            raw.instance = this.instance;
            raw.client = creator.create(instance, timeout);
            return raw;
        }

        public override IList<RawClientInfo> getRowClients()
        {
            IList<RawClientInfo> infos = new List<RawClientInfo>();
            RawClientInfo raw = new RawClientInfo();
            raw.instance = this.instance;
            raw.client = creator.create(raw.instance, timeout);
            infos.Add(raw);
            return infos;
        }
    }

}
